README.md

```diff
@@ -464,10 +464,77 @@ progress = (
 </tr>
 </table>
 
+#### Regular Expressions
+
+It can be useful to sort an entire group based on a non-prefix substring. The
+option `by_regex=…` takes a comma-separated list of [regular
+expressions][regex-sytax] that will be applied to the group, and then sorting
+will take place on just the results of the regular expressions.
+
+> [!TIP]
+> Regular expressions often need special characters. See [Syntax](#syntax) below
+> for how to include special characters in the `by_regex` option.
+
+By default, all characters that the regular expression matches will be
+considered for sorting. If the regular expression contains any capturing groups,
+only the characters matched by the capturing groups will be considered for
+sorting. The result from each regular expression will be concatenated into a
+list of results, and that list of results will be sorted [lexicographically].
+
+Regular expressions are applied **before** other sorting options, so
+[`prefix_order`](#prefix-sorting) will only apply to the characters matched by
+your regular expressions.
+
+[regex-syntax]: http://godoc/pkg/regexp/syntax/
+[lexicographically]: https://en.wikipedia.org/wiki/Lexicographic_order
+
+<table border="0">
+<tr>
+<td>
+
+```java
+# keep-sorted start
+List<String> foo;
+String bar;
+Object baz;
+# keep-sorted end
+```
+
+```java
+# keep-sorted start
+List<String> foo;
+String bar;
+Object baz;
+# keep-sorted end
+```
+
+</td>
+<td>
+
+```diff
++# keep-sorted start by_regex=\b.+;\b
+ String bar;
+ Object baz;
+ List<String> foo;
+ # keep-sorted end
+```
+
+```diff
++# keep-sorted start by_regex=\b.+;\b prefix_order=foo
```

Comment:
It's not clear from the doc for prefix_order that `prefix_order=foo` is equivalent to `prefix_order=foo,`, i.e. that if no empty prefix is explicitly specified, one is assumed to be at the end. I can infer that from this example, but perhaps it's worth mentioning  explicitly